bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateService.java

```diff
@@ -199,6 +199,17 @@ public Void call() {
             }
         });
     }
+    
+    public void cancelFirmwareUpdate(final ThingUID thingUID){
+        Preconditions.checkNotNull(thingUID, "Thing UID must not be null.");
+        final FirmwareUpdateHandler firmwareUpdateHandler = getFirmwareUpdateHandler(thingUID);
```

Comment:
-updateFirmware throws an IllegalStateException
-updateFirmware and cancelFirmwareUpdate should use the same code for firmware update handler check